src/app/pages/dashboard/job-result/job-result.component.ts

```diff
@@ -220,6 +223,9 @@ export class JobResultComponent implements OnInit, OnDestroy {
       ? event.data.report
       : event.data.errors;
     this.editor.update(json);
+
+    if ('screenshot' in json && json['screenshot'].length)
```

Comment:
using the `Object.prototype.hasOwnProperty` to check for key would be a more "javascript" approach, I believe.